osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementService.java

```diff
@@ -469,6 +347,198 @@ public Organisation findOrganisation(@Identification final String organisationId
+                    this.deviceSpecifications.isManagedExternally(deviceFilter.getDeviceExternalManaged().getValue()));
+        }
+        return specifications;
+    }
+
+    private Specifications<Device> doFilterOnAddress(final DeviceFilter deviceFilter,
+            Specifications<Device> specifications) throws ArgumentNullOrEmptyException {
+        if (!StringUtils.isEmpty(deviceFilter.getCity())) {
+            specifications = specifications
+                    .and(this.deviceSpecifications.hasCity(deviceFilter.getCity().replaceAll(WILDCARD, "%") + "%"));
+        }
+        if (!StringUtils.isEmpty(deviceFilter.getPostalCode())) {
+            specifications = specifications.and(this.deviceSpecifications
+                    .hasPostalCode(deviceFilter.getPostalCode().replaceAll(WILDCARD, "%") + "%"));
+        }
+        if (!StringUtils.isEmpty(deviceFilter.getStreet())) {
+            specifications = specifications
+                    .and(this.deviceSpecifications.hasStreet(deviceFilter.getStreet().replaceAll(WILDCARD, "%") + "%"));
+        }
+        if (!StringUtils.isEmpty(deviceFilter.getNumber())) {
+            specifications = specifications
+                    .and(this.deviceSpecifications.hasNumber(deviceFilter.getNumber().replaceAll(WILDCARD, "%") + "%"));
+        }
+        if (!StringUtils.isEmpty(deviceFilter.getMunicipality())) {
+            specifications = specifications.and(this.deviceSpecifications
+                    .hasMunicipality(deviceFilter.getMunicipality().replaceAll(WILDCARD, "%") + "%"));
+        }
+        return specifications;
+    }
+
+    private Specifications<Device> doFilterOnDeviceAlias(final DeviceFilter deviceFilter,
+            Specifications<Device> specifications) throws ArgumentNullOrEmptyException {
+        if (!StringUtils.isEmpty(deviceFilter.getAlias())) {
+            specifications = specifications
+                    .and(this.deviceSpecifications.hasAlias(deviceFilter.getAlias().replaceAll(WILDCARD, "%") + "%"));
+        }
+        return specifications;
+    }
+
+    private Specifications<Device> doFilterOnDeviceIdentification(final DeviceFilter deviceFilter,
+            Specifications<Device> specifications) throws ArgumentNullOrEmptyException {
+        if (!StringUtils.isEmpty(deviceFilter.getDeviceIdentification())) {
+            String searchString = deviceFilter.getDeviceIdentification();
+
+            if (!deviceFilter.isExactMatch()) {
+                searchString = searchString.replaceAll(WILDCARD, "%") + "%";
+            }
+
+            specifications = specifications
+                    .and(this.deviceSpecifications.hasDeviceIdentification(searchString, deviceFilter.isExactMatch()));
+        }
+        return specifications;
+    }
+
+    private Specifications<Device> doFilterOnOrganisationIdentification(final DeviceFilter deviceFilter,
+            final Organisation organisation) throws FunctionalException, ArgumentNullOrEmptyException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)